core/callitopservice.class.inc.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/**
+ * @since 1.3.0
+ */
+class CallItopService
+{
+	public function CallItopViaHttp($sUri, $aAdditionalData, $iTimeOut = -1)
+	{
+
+		$sUrl = Utils::GetConfigurationValue('itop_url', '').$sUri;
+
+
+		$aData = array_merge(
+			array(
+				'auth_user' => Utils::GetConfigurationValue('itop_login', ''),
+				'auth_pwd' => Utils::GetConfigurationValue('itop_password', ''),
+			),
+			$aAdditionalData
+		);
+
+		// timeout in seconds, for a synchro to run
+		$iCurrentTimeOut = ($iTimeOut === -1) ? (int)Utils::GetConfigurationValue('itop_synchro_timeout', 600) : $iTimeOut;
+		$aRawCurlOptions = Utils::GetConfigurationValue('curl_options', [CURLOPT_SSLVERSION => CURL_SSLVERSION_SSLv3]);
+		$aCurlOptions = $this->GetCurlOptions($aRawCurlOptions, $iCurrentTimeOut);
```

Comment:
This might be better located in Utils? The naming is also not ideal.
```suggestion
		$aCurlOptions = Utils::ProcessCurlOptions($aRawCurlOptions, $iCurrentTimeOut);
```

Reply:
+1 i did it and reused added method